_overviews/scala3-book/taste-contextual-abstractions.md

```diff
@@ -12,19 +12,29 @@ next-page: taste-toplevel-definitions
 TODO: Now that this is a separate section, it needs a little more content.
 {% endcomment %}
 
+Under certain circumstances, you can omit some parameters of method calls that are considered repetitive.
 
-Under certain circumstances, the Scala compiler can “write” some parts of your programs.
-For instance, consider a program that sorts a list of addresses by two criteria, city name and then street name:
+Those parameters are called _Context Parameters_ because there are inferred by the compiler from the context surrounding the method call.
+
+For instance, consider a program that sorts a list of addresses by two criteria: the city name and then street name.
 
 ```scala
 val addresses: List[Address] = ...
+
+addresses.sortBy(address => (address.city, address.street))(using Ordering.Tuple2(Ordering.String, Ordering.String))
 addresses.sortBy(address => (address.city, address.street))
```

Comment:
What do you think of showing first the normal Scala version (with no explicit `Ordering` parameter), and then explain that the compiler infers the `Ordering` parameter to `Ordering.Tuple2(Ordering.String, Ordering.String)`?

Reply:
I totally agree!